File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    // 중복 제거: Set 사용
    const kinds = new Set(nums);
    
    // 가져갈 수 있는 수
    const maxPick = nums.length / 2;
    
    // 최대 종류 수 = min(서로 다른 종류 수, 가져갈 수 있는 수)
    return Math.min(kinds.size, maxPick);
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    # 고를 수 있는 최대 마리 수
    pick = len(nums) // 2
    # 서로 다른 폰켓몬 종류 수
    kinds = len(set(nums))
    # 둘 중 작은 값이 최대 종류 수
    return min(pick, kinds)
